### PR TITLE
Fixes certain objects not affecting contents inside when hit by explosions

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[bugfix][consistency]

## What this does
takes the bugfixes from #34153, along with its relevant code refactors

## Changelog
:cl:
 * bugfix: Exploded body scanners, sleepers, cloning pods, morgue trays and cremators now properly affect mobs inside when destroyed.
 * tweak: Racks no longer have their parts more likely to be destroyed in low intensity explosions than medium ones.

### CODE REVIEW

**Comment by Eneocho** [comment]
image

**Review by jwhitak — APPROVED** [review]
I'm gonna be real here, I kinda prefer the switch method in terms of code readability. The severity to my knowledge is always 1, 2, or 3, representing DEV, HEAVY, and LIGHT respectively. Throwing those arbitrary DEFINE numbers into a math formula doesn't feel right.

But that said, the rest of the code checks out. I'm actually surprised that exploding a sleeper doesn't infinite loop.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -331,42 +331,19 @@
...
-				if(has_electronics)
-					E = dump_electronics()
-					E.forceMove(src)
-				for(var/atom/movable/A as mob|obj in src)
-					A.forceMove(src.loc)
-					A.ex_act(severity++)
-				dump_contents()
-				qdel(src)
+	var/probdivide = severity == 3 ? 20 : severity
+	if(prob(100/probdivide)) //1 = 100, 2 = 50, 3 = 5
```
Comment (CrazyAmphibian) [inline code/game/objects/structures/crates_lockers/closets.dm:337]: i don't think this'll improve performance at all, and makes the code harder to read.
i don't get the point of doing this.

Comment (SECBATON-GRIFFON) [reply code/game/objects/structures/crates_lockers/closets.dm:337]: just wanted to cut down on copy-pasted-repetition

Comment (CrazyAmphibian) [reply code/game/objects/structures/crates_lockers/closets.dm:337]: i mean, there's still a lot of copy-paste

**Comment by KingofHONK** [comment]
Why not do a "Yes, but" for this one? Sure the machine could get destroyed but instead of complete immunity to the explosion, maybe break some bones and wound them instead. Bomb dodging is fun and rewards players for being quick on their feet, and removing one of the few (if not only) counterplays to explosions is lame

**Comment by SECBATON-GRIFFON** [comment]
clearer lack of support

**Pull request closed without merging**